diesel/src/mysql/types/mod.rs

```diff
@@ -1,12 +1,19 @@
 #[cfg(feature = "chrono")]
 mod date_and_time;
+pub mod numeric;
 
 use byteorder::WriteBytesExt;
-use mysql::{Mysql, MysqlType};
+pub use mysql::{Mysql, MysqlType};
 use std::error::Error as StdError;
 use std::io::Write;
 use types::{ToSql, IsNull, FromSql, HasSqlType};
 
+pub mod sql_types {
```

Comment:
We don't reexport `BigDecimal` for postgres, why should we for MySQL ?

I think you can remove the whole `sql_types` module :)